src/rockstor/storageadmin/static/storageadmin/js/router.js

```diff
@@ -757,7 +758,30 @@ var AppRouter = Backbone.Router.extend({
         this.currentLayout = new LogsView();
         $('#maincontent').empty();
         $('#maincontent').append(this.currentLayout.render().el);
-    }
+    },
+	
+	showShell: function() {
+		//Special router function for shell
+		//We fetch shellinaboxd service model and check for detach option
+		//If not present or false we go with normal layout to maincontent
+		//otherwhise (popup enabled) we open it in a new detached window
+		var _this = this;
+		_this.shell_service = new Service({name: 'shellinaboxd'});
+		_this.shell_service.fetch({
+		success: function(collection){
+			var config = JSON.parse(collection.get('config'));
+			if ('detach' in config && config.detach){
```

Comment:
Thinking about current first run of shellinaboxd we have some defaults settings for it (black on white and LOGIN connection as on shellinaboxd conf file). What do you think @schakrava about a prep_db.py change to full reflect defaults vals? Changing / adding a new func next to 
```python
def register_services():